pkg/controller/nodemaintenance/nodemaintenance_controller_test.go

```diff
@@ -249,3 +259,266 @@ var _ = Describe("updateCondition", func() {
+				Groups: []metav1.APIGroup{{
+					Name: "lease",
+					Versions: []metav1.GroupVersionForDiscovery{{
+						GroupVersion: "chaos.k8s.io/v1beta1",
+						Version:      "v1beta1",
+					}},
+				}},
+			}}}
+		}
+
+		var objs []runtime.Object
+
+		nm, objs = getCommonTestObjs()
+
+		clObjs := append(objs, nm)
+		cl = fake.NewFakeClient(clObjs...)
+		cs = k8sfakeclient.NewSimpleClientset(objs...)
+	}
+
+	checkLeaseEstablished := func()  {
+		nName := types.NamespacedName{Namespace: LeaseNamespace, Name: "node01"}
+		lease := &coordv1beta1.Lease{}
+
+		err := cl.Get(context.TODO(), nName, lease)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(*lease.Spec.HolderIdentity).To(Equal(LeaseHolderIdentity))
+	}
+
+	checkLeaseNotEstablished := func()  {
+		nName := types.NamespacedName{Namespace: LeaseNamespace, Name: "node01"}
+		lease := &coordv1beta1.Lease{}
+
+		err := cl.Get(context.TODO(), nName, lease)
+		Expect(err).To(HaveOccurred())
+	}
+
+	checkSuccesfulReconcile := func(withLease bool) {
+		maintanance := &kubevirtv1alpha1.NodeMaintenance{}
+		err := cl.Get(context.TODO(), req.NamespacedName, maintanance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(maintanance.Status.Phase).To(Equal(kubevirtv1alpha1.MaintenanceSucceeded))
+
+		if withLease {
+			checkLeaseEstablished()
+		} else {
+			checkLeaseNotEstablished()
+		}
+	}
+
+	checkFailedReconcile := func() {
+		maintanance := &kubevirtv1alpha1.NodeMaintenance{}
+		err := cl.Get(context.TODO(), req.NamespacedName, maintanance)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(maintanance.Status.LastError)).NotTo(Equal(0))
+
+		checkLeaseNotEstablished()
+	}
+
+	reconcileMaintenance := func(nm *kubevirtv1alpha1.NodeMaintenance) {
+		// Mock request to simulate Reconcile() being called on an event for a
+		// watched resource .
+		req = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name: nm.ObjectMeta.Name,
+			},
+		}
+		r.Reconcile(req)
+	}
+
+	taintExist := func(node *corev1.Node, key string, effect corev1.TaintEffect) bool {
+		checkTaint := corev1.Taint{
+			Key:    key,
+			Effect: effect,
+		}
+		taints := node.Spec.Taints
+		for _, taint := range taints {
+			if reflect.DeepEqual(taint, checkTaint) {
+				return true
+			}
+		}
+		return false
+	}
+
+	setupTest:=func(withLease bool) {
+
+		ctrl = gomock.NewController(GinkgoT())
+		mockMaintenanceReconcile = NewMockReconcileHandler(ctrl)
+		Handler = mockMaintenanceReconcile
+
+		s := k8scheme.Scheme
+		s.AddKnownTypes(kubevirtv1alpha1.SchemeGroupVersion, nm)
+
+		setFakeClients(withLease)
+
+		kubeSharedInformer := informers.NewSharedInformerFactoryWithOptions(cs, 2*time.Minute)
+		fakePodInformer := kubeSharedInformer.Core().V1().Pods()
+
+		// Create a ReconcileNodeMaintenance object with the scheme and fake client
+		r = &ReconcileNodeMaintenance{client: cl, scheme: s, podInformer: fakePodInformer.Informer()}
+		initDrainer(r, fcd)
```

Comment:
you also have initDrainer in beforeEach().  I suggest that you check if there are no double invocations in setupTest vs beforeEach functions.

Reply:
the test Describe("updateConditionWithLease") doesn't have a beforeEach block.